llpc/tool/amdllpc.cpp

```diff
@@ -610,6 +659,23 @@ static Error processInputs(ICompiler *compiler, InputSpecGroup &inputSpecs) {
 #endif
   }
 
+  RtState *rtState = nullptr;
+  switch (compileInfo.pipelineType) {
+  case VfxPipelineTypeGraphics:
+    rtState = &compileInfo.gfxPipelineInfo.rtState;
+    break;
+  case VfxPipelineTypeCompute:
+    rtState = &compileInfo.compPipelineInfo.rtState;
+    break;
+  case VfxPipelineTypeRayTracing:
+    rtState = &compileInfo.rayTracePipelineInfo.rtState;
+    break;
+  }
+
+  std::vector<char> gpurtShaderLibraryStorage;
+  if (Error err = fixupRtState(*rtState, gpurtShaderLibraryStorage))
```

Comment:
Maybe we can let the gpurt lib storage be owned by the `rtState` object, and not worry about the lifetime of the storage? Or do we need to copy `rtState` objects around?

Reply:
Unfortunately that doesn't work because `Vkgc::RtState` is part of the driver/compiler interface that is sometimes treated in a more C-like fashion.